### PR TITLE
Add --force option to "roxctl image check"

### PR DESCRIPTION
**Pull request by srcporter**
## Description

Following the example in roxctl/image/scan/scan.go, added a "--force" option to roxctl image check, to force Central to ignore its cache for the image and force a re-pull from Scanner 

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

With the new version of roxctl, tested with and without the "force" option in image check and observed logs in Central and Scanner. Using an image that is already known to Central:

roxctl -e $CENTRAL:443 image check --insecure-skip-tls-verify --image quay.io/rhacs-misc/log4shell:1.0

Results in no new log lines in Central

With the same image:

roxctl -e $CENTRAL:443 image check --force --insecure-skip-tls-verify --image quay.io/rhacs-misc/log4shell:1.0

Results in this new log line:
pkg/images/enricher: 2023/02/07 19:44:09.492138 enricher_impl.go:264: Info: Getting metadata for image quay.io/rhacs-misc/log4shell:1.0

### CODE REVIEW

File: roxctl/image/check/check.go [hunk 1]
```diff
@@ -85,6 +85,7 @@ func Command(cliEnvironment environment.Environment) *cobra.Command {
 	pkgUtils.Must(c.MarkFlagRequired("image"))
 	c.Flags().IntVarP(&imageCheckCmd.retryDelay, "retry-delay", "d", 3, "set time to wait between retries in seconds.")
 	c.Flags().IntVarP(&imageCheckCmd.retryCount, "retries", "r", 3, "number of retries before exiting as error.")
+	c.Flags().BoolVarP(&imageCheckCmd.force, "force", "f", false, "the --force flag ignores Central's cache for the check and forces a fresh re-pull from Scanner")
```
Comment (connorgorman) [inline roxctl/image/check/check.go:88]: Wording: cache for the image scan and instead forces a refresh from Scanner. Maybe something like that. Just describing which cache it's ignoring

File: roxctl/image/check/check.go [hunk 2]
```diff
@@ -295,14 +297,15 @@ func printAdditionalWarnsAndErrs(numTotalViolatedPolicies int, results []policy.
 }
 
 // Use inputs to generate an image name for request.
-func buildRequest(image string, sendNotifications bool, policyCategories []string) (*v1.BuildDetectionRequest, error) {
+func buildRequest(image string, sendNotifications bool, force bool, policyCategories []string) (*v1.BuildDetectionRequest, error) {
```
Comment (connorgorman) [inline roxctl/image/check/check.go:300]: `sendNotifications, force bool`

File: roxctl/image/check/check.go [hunk 3]
```diff
@@ -295,14 +297,15 @@ func printAdditionalWarnsAndErrs(numTotalViolatedPolicies int, results []policy.
...
 	img, err := utils.GenerateImageFromString(image)
 	if err != nil {
 		return nil, errors.Wrapf(err, "could not parse image '%s'", image)
 	}
 	return &v1.BuildDetectionRequest{
 		Resource:          &v1.BuildDetectionRequest_Image{Image: img},
 		SendNotifications: sendNotifications,
+        Force: force,
```
Comment (connorgorman) [inline roxctl/image/check/check.go:308]: formatting looks weird here

File: roxctl/image/check/check.go (hunk 1 above)
Comment (srcporter) [reply roxctl/image/check/check.go]: Fixed in latest commit

File: roxctl/image/check/check.go (hunk 2 above)
Comment (srcporter) [reply roxctl/image/check/check.go]: Fixed in latest commit

File: roxctl/image/check/check.go (hunk 3 above)
Comment (srcporter) [reply roxctl/image/check/check.go]: Fixed in latest commit

File: central/detection/service/service_impl.go [hunk 4]
```diff
@@ -136,9 +136,15 @@ func (s *serviceImpl) DetectBuildTime(ctx context.Context, req *apiV1.BuildDetec
 	img := types.ToImage(image)
 
 	enrichmentContext := enricher.EnrichmentContext{}
+	if req.GetNoExternalMetadata() && req.GetForce() {
```
Comment (connorgorman) [inline central/detection/service/service_impl.go:139]: Nit: GetNoExternalMetadata is a purely code construct, we should describe what that means. e.g. incompatible with not fetching metadata from external sources

File: roxctl/image/check/check.go [hunk 5]
```diff
@@ -85,6 +85,7 @@ func Command(cliEnvironment environment.Environment) *cobra.Command {
 	pkgUtils.Must(c.MarkFlagRequired("image"))
 	c.Flags().IntVarP(&imageCheckCmd.retryDelay, "retry-delay", "d", 3, "set time to wait between retries in seconds.")
 	c.Flags().IntVarP(&imageCheckCmd.retryCount, "retries", "r", 3, "number of retries before exiting as error.")
+	c.Flags().BoolVarP(&imageCheckCmd.force, "force", "f", false, "the --force flag ignores Central's cache for the image and forces a fresh re-pull from Scanner")
```
Comment (connorgorman) [inline roxctl/image/check/check.go:88]: nit: ignores -> bypasses

Comment (srcporter) [reply roxctl/image/check/check.go]: fixed in latest commit

File: roxctl/image/check/check.go [hunk 6]
```diff
@@ -85,6 +85,7 @@ func Command(cliEnvironment environment.Environment) *cobra.Command {
 	pkgUtils.Must(c.MarkFlagRequired("image"))
 	c.Flags().IntVarP(&imageCheckCmd.retryDelay, "retry-delay", "d", 3, "set time to wait between retries in seconds.")
 	c.Flags().IntVarP(&imageCheckCmd.retryCount, "retries", "r", 3, "number of retries before exiting as error.")
+	c.Flags().BoolVarP(&imageCheckCmd.force, "force", "f", false, "the --force flag bypasses Central's cache for the image and forces a fresh re-pull from Scanner")
```
Comment (0x656b694d) [inline roxctl/image/check/check.go:88]:
```suggestion
	c.Flags().BoolVarP(&imageCheckCmd.force, "force", "f", false, "bypass Central's cache for the image and force a new pull from the Scanner")
```
?

**Review by dhaus67 — APPROVED** [review]

**Comment by gavin-stackrox** [comment]
Hey @srcporter, can you resubmit this PR from a branch directly off github.com/stackrox/stackrox? It will make it much easier to run builds and CI. Thanks.

**Review by 0x656b694d — APPROVED** [review]

**Pull request merged**